### PR TITLE
add dev container and tests

### LINKED ISSUE #10: Create GitHub workflow to automate tests for this Repo
We need workflows to test changes in this repo.
The test is to validate that the code of this repo works correctly.

You may reference the tests implemented at the main [render engine](https://github.com/render-engine/render-engine) repo to help you.

We only need to create tests here just like the ones on the main repo.
If you can create new tests that would be very helpful for us!

### LINKED ISSUE #18: Test Generation of the site
We've had issues in the past where generation was not correct and we didn't detect until a new version was released. 

We should build this and test that things build.

### PR DESCRIPTION
**Pull request by john0isaac**
This PR adds DevContainer support following our [render-engine/render_engine](https://github.com/render-engine/render_engine) configuration in addition to setting up testing for the site generation with Cookiecutter and site build with render-engine-cli.

fix #18
fix #10

### CODE REVIEW

**Comment by john0isaac** [comment]
@kjaymiller I think render-engine is not compatible with python 3.9 😄 

```
  File "/tmp/pytest-of-runner/pytest-0/test_site_generation0/.venv/lib/python3.9/site-packages/render_engine/page.py", line 29, in BasePage
    template: str | type[jinja2.Template] | None
TypeError: unsupported operand type(s) for |: 'type' and 'types.GenericAlias'
```

File: tests/test_template.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    install_args = split(f"{python} -m {install_cli}")
+    install_process = run(install_args)
+    assert install_process.returncode == 0
+
+
+def test_site_generation(context, project, python, install_render_engine_cli):
+    generate_site = "render_engine build app:app"
+    generate_args = split(f"{python} -m {generate_site}")
+    generate_process = run(generate_args, cwd=project)
+    assert generate_process.returncode == 0
```
Comment (john0isaac) [inline tests/test_template.py:82]: we can add more tests here as we see fit in the future
for now I'm testing that the static site is generated and built successfully.

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,5 @@
+ARG IMAGE="python:3.12"
```
Comment (kjaymiller) [inline .devcontainer/Dockerfile:1]: any reason not 3.13?

**Review by kjaymiller — APPROVED** [review]
one minor comment but this looks good. Free to merge!

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (john0isaac) [reply .devcontainer/Dockerfile:1]: we did discuss this before it was not available at the time of our discussion.
let me check if it's available now

Comment (john0isaac) [reply .devcontainer/Dockerfile:1]: I think it's available now
https://github.com/devcontainers/images/tree/main/src/python 

let me test it tonight.

Comment (john0isaac) [reply .devcontainer/Dockerfile:1]: they are lying the only available 3.13 is dev-3.13.
let's keep 3.12 until they release a stable 3.13

See:
https://mcr.microsoft.com/v2/vscode/devcontainers/python/tags/list

**Pull request merged**